### ISSUE TITLE
Static Web App Deployment issue

### ISSUE THREAD

**Issue by parvam6** [FBoucher/TinyBlazorAdmin#37]
@FBoucher when i try to create a static web app and Deploy TinyBlazorAdmin (the Frontend) the build fails and i am getting the error 

![image](https://user-images.githubusercontent.com/77300381/104358204-19f5a780-54dc-11eb-99b3-e246460db4b2.png)

Is there a  way to resolve the issue.?

**Comment by scarpentier** [comment]
I think I have the same issue. I don't get why the compiler has trouble finding System.Text.Encodings.Web (5.0.0).

Could it be because the refered assembly is strongly typed to 5.0.0 and the .NET 5 version installed is 5.0.100 ?

```
---Oryx build logs---


Operation performed by Microsoft Oryx, https://github.com/Microsoft/Oryx
You can report issues at https://github.com/Microsoft/Oryx/issues

Oryx Version: 0.2.20201112.1, Commit: a120bb248aa8d208ca5e02d7d9d075196e79cffd, ReleaseTagName: 20201112.1

Build Operation ID: |431yFGCCE8s=.e7c3995f_
Repository Commit : f0e2c66be628210ed1fd1330756a749a7fd946a6

Detecting platforms...
Detected following platforms:
  dotnet: 5.0.0
Version '5.0.0' of platform 'dotnet' is not installed. Generating script to install it...


Source directory     : /github/workspace/src/TinyBlazorAdmin
Destination directory: /bin/staticsites/ss-oryx/app


Downloading and extracting 'dotnet' version '5.0.100' to '/tmp/oryx/platforms/dotnet/5.0.100'...
Downloaded in 2 sec(s).
Verifying checksum...
Extracting contents...
Done in 7 sec(s).


Using .NET Core SDK Version: 5.0.100

Welcome to .NET 5.0!
---------------------
SDK Version: 5.0.100

Telemetry
---------
The .NET tools collect usage data in order to help us improve your experience. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Installed an ASP.NET Core HTTPS development certificate.
To trust the certificate run 'dotnet dev-certs https --trust' (Windows and macOS only).
Learn about HTTPS: https://aka.ms/dotnet-https
----------------
Write your first app: https://aka.ms/dotnet-hello-world
Find out what's new: https://aka.ms/dotnet-whats-new
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------
  Determining projects to restore...
  Restored /github/workspace/src/TinyBlazorAdmin/TinyBlazorAdmin.csproj (in 2.59 sec).

Publishing to directory /bin/staticsites/ss-oryx/app...

Microsoft (R) Build Engine version 16.8.0+126527ff1 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  Fatal error in Mono IL Linker
/var/nuget/microsoft.aspnetcore.components.webassembly.build/3.2.1/targets/Blazor.MonoRuntime.targets(326,5): error : Unhandled exception. Mono.Linker.Steps.XmlResolutionException: Failed to process XML description: <unspecified> [/github/workspace/src/TinyBlazorAdmin/TinyBlazorAdmin.csproj]
   ---> Mono.Cecil.AssemblyResolutionException: **Failed to resolve assembly: 'System.Text.Encodings.Web, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'**
   ---> System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
     at Mono.Collections.Generic.Collection`1.get_Item(Int32 index)
     at Mono.Cecil.BaseAssemblyResolver.GetAssemblyInNetGac(AssemblyNameReference reference, ReaderParameters parameters)
     at Mono.Cecil.BaseAssemblyResolver.GetAssemblyInGac(AssemblyNameReference reference, ReaderParameters parameters)
     at Mono.Cecil.BaseAssemblyResolver.Resolve(AssemblyNameReference name, ReaderParameters parameters)
     at Mono.Linker.AssemblyResolver.Resolve(AssemblyNameReference name, ReaderParameters parameters)
     at Mono.Linker.LinkContext.Resolve(IMetadataScope scope)
     at Mono.Linker.LinkContext.Resolve(IMetadataScope scope)
     at Mono.Linker.LinkContext.ResolveReferences(AssemblyDefinition assembly)
     at Mono.Linker.Steps.ResolveFromXmlStep.ProcessAssemblies(LinkContext context, XPathNodeIterator iterator)
     at Mono.Linker.Steps.ResolveFromXmlStep.Process()
     --- End of inner exception stack trace ---
     at Mono.Linker.Steps.ResolveFromXmlStep.Process()
     at Mono.Linker.Steps.BaseStep.Process(LinkContext context)
     at Mono.Linker.Pipeline.ProcessStep(LinkContext context, IStep step)
     at Mono.Linker.Pipeline.Process(LinkContext context)
     at Mono.Linker.Driver.Run(ILogger customLogger)
     at Mono.Linker.Driver.Execute(String[] args, ILogger customLogger)
     at Mono.Linker.Driver.Main(String[] args)
/var/nuget/microsoft.aspnetcore.components.webassembly.build/3.2.1/targets/Blazor.MonoRuntime.targets(326,5): error : ILLink failed with exit code 134. [/github/workspace/src/TinyBlazorAdmin/TinyBlazorAdmin.csproj]


---End of Oryx build logs---
```

**Comment by scarpentier** [comment]
After some tinkering I was able to make it work by reverting the project to `netstandard2.1` instead of `net5.0`.

.NET 5 looks fun but it seems Oryx is not ready for it just yet.

Here's what my TinyBlazorAdmin.csproj file looks like:

```
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netstandard2.1</TargetFramework>
    <RazorLangVersion>3.0</RazorLangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.Blazor.HttpClient" Version="3.2.0-preview3.20168.3" />
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="3.2.1" />
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Build" Version="3.2.1" PrivateAssets="all" />
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="3.2.1" PrivateAssets="all" />
    <PackageReference Include="Microsoft.Authentication.WebAssembly.Msal" Version="3.2.1" />
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="5.0.0-preview.4.20251.6" />
    <PackageReference Include="Microsoft.Extensions.Http" Version="5.0.0-preview.3.20215.2" />
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
    <PackageReference Include="System.Net.Http.Json" Version="3.2.1" />
  </ItemGroup>

</Project>
```

**Comment by FBoucher** [comment]
the version for dependencies are fixed with PR #46
